lib/rust/profiler/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+use std::cell;
+use std::num;
+use std::str;
+
+
+
+// =============
+// === Label ===
+// =============
+
+/// The label of a profiler; this includes the name given at its creation, along with file and
+/// line-number information.
+pub type Label = &'static str;
+
+
+
+// =================
+// === Timestamp ===
+// =================
+
+/// Time elapsed since the time origin (https://www.w3.org/TR/hr-time-2/#sec-time-origin).
+///
+/// Stored in units of 100us, because that is maximum resolution of performance.now():
+/// - in specification, 100us is the limit: https://www.w3.org/TR/hr-time-3
+/// - in practice, as observed in debug consoles: Chromium 97 (100us) and Firefox 95 (1ms)
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct Timestamp(num::NonZeroU64);
+
+impl Timestamp {
+    /// Return the current time, relative to the time origin.
+    pub fn now() -> Self {
+        Self::from_ms(js::performance::now())
+    }
+
+    /// Return the timestamp corresponding to an offset from the time origin, in ms.
+    pub fn from_ms(ms: f64) -> Self {
+        let ticks = (ms * 10.0).round() as u64;
+        // Safety: ticks + 1 will not be 0 unless a Timestamp wraps.
+        // It takes (2 ** 64) * 100us = 58_455_453 years for a Timestamp to wrap.
+        unsafe { Self(num::NonZeroU64::new_unchecked(ticks + 1)) }
+    }
+
+    /// Convert to an offset from the time origin, in ms.
+    pub fn into_ms(self) -> f64 {
+        (self.0.get() - 1) as f64 / 10.0
```

Comment:
in both equations you are using the a little bit magical `+ 1` and `- 1`. Could you extract the `1` constnt to a well named constant that describes its semantics, please?